### ISSUE TITLE
LUX Interrogation for Circle 2

### ISSUE THREAD

**Issue by whreams** [klanchman/homebridge-logi-circle-controls#22]
Now that the circle 2 firmware supports LUX readings, any chance the plugin can be extended to read the current LUX level?  The native HK plugin currently does not support this and multiple attempts to ask Logitech to support have gone unanswered.  TIA! -  Bill

**Comment by klanchman** [comment]
Thanks for the suggestion!

It looks like readouts from the ambient light sensor were only added to the HomeKit Secure Video firmware, not the "normal" firmware. As mentioned in #19, this plugin is not compatible with the HKSV firmware.

It looks like the official public API does not mention anything about the ambient light sensor (I'm planning to switch to use this API eventually). However, the internal API does have a field called `alsLevel`. However, the value never seems to change on my camera.

I'll keep an eye on this, but currently it looks like adding this feature is not possible.

**Comment by whreams** [comment]
Thanks for the response!
Any chance you can share the API code so I could test on my camera?
To note, the Eve app does have the LUX value in it's mobile app - so I believe somehow it can be interrogated.

**Comment by klanchman** [comment]
> Any chance you can share the API code so I could test on my camera?

Assuming you are not using HKSV firmware, you can log into https://circle.logi.com on a desktop browser with your developer tools open to the Network tab. Wait for the page to show a feed from your camera, then go to the address bar and copy the UUID in the URL (the part after `accessories/`) and copy it. Paste that UUID into the search box of the Network tool, and find a request that has that UUID in it. It should be a `GET` request to `/api/accessories/:id`. From there, you can open the response and search for `alsLevel` to find that property.

> To note, the Eve app does have the LUX value in it's mobile app - so I believe somehow it can be interrogated.

I checked Eve last night and did not see any ambient light readings for my camera. However, I am not on the HKSV firmware. Maybe if I was on the HKSV firmware it'd show up?

***

In checking it again, it looks like the value in the internal API _does_ actually change, it just may not have been sensitive enough to pick up the changes in lighting I was trying to test with yesterday. I'll play around with it some more to get an idea of how sensitive it is, and to try to ensure it's measuring in lux and not some other unit.

I'd still like to see this value show up in the [public API](https://developers.logitech.com/circle) since I plan to migrate to that. I may need to reach out to Logitech to see about getting that added.

**Comment by whreams** [comment]
Awesome.  Happy to help test in any way...

**Comment by whreams** [comment]
Any chance to access the LUX reading and confirm it's use in the plugin?

**Comment by klanchman** [comment]
Right now my main priorities for new development are migrating to TypeScript (#18) and the public API (#17).

The public API does not make the ambient light reading available, only Logitech's internal API appears to. I'm reluctant to add support for anything that the public API does not support, which is why I tagged this feature as "blocked."

I'm working on getting OAuth credentials so I can access the public API. While doing that, I'll also ask the team at Logitech if it's possible for them to add ambient light readings to the public API.

In other words, this feature is on my radar, but it's not something that I'll be able to do soon 😅

**Comment by whreams** [comment]
Ok.  Thanks for the response!

**Comment by klanchman** [comment]
Some good news and bad news here.

Bad news: I haven't heard back about access to the public API yet. I go into more detail in [my comment on #17](https://github.com/klanchman/homebridge-logi-circle-controls/issues/17#issuecomment-782768831)), but suffice to say I don't expect to migrate off the internal API for a while.

Good news: Since I don't expect to leave the internal API for a while, I'll probably look at this soon. 🙂

@whreams what were you looking for out of the ambient light reading? Did you just want to see it in the Home app, or did you want to be able to do automations when it changes? I ask because just seeing it should be straightforward enough, but doing automations would be tough.

**Comment by whreams** [comment]
Good stuff. So I wanted to be able to read the level so I could automate off of it. Ideally be able to read the Liz reading and then turn on and off additional accessories via node red.

**Comment by klanchman** [comment]
Gotcha, I think that makes sense 👍 
It sounds like you'd be checking the sensor periodically from Node-RED, so from this plugin's perspective the sensor would behave like any of the other switches. (The sensor would only update when asked, rather than periodically.)

I'll caution that since the plugin needs to contact Logitech's servers to read the state of the sensor / switches / etc., you won't want to check the value too frequently, maybe just every few minutes or so 😄

**Comment by whreams** [comment]
Agree. No different than myq, or any other plugin that relys on api availability. Thanks for remembering and for the consideration.

**Comment by klanchman** [comment]
I've been working on and off on the next major version of the plugin, which I'm planning to include ambient light sensor support in.

My camera doesn't seem to update its ambient light reading very often. I'm not sure if this is expected, or if there's something funky with my camera.

Regardless, there's a beta version of the plugin for you to try. You can install it like so:

```
npm i -g homebridge-logi-circle-controls@next
```

Note that Node v12 is required to run the beta version of the plugin, so you may need to upgrade Node for this to work.

Play around with it for a little while and let me know how it works for you. Hopefully better than it works for me! 😅
